CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ cmake_minimum_required(VERSION 3.12)
 set(CMAKE_DISABLE_SOURCE_CHANGES ON)
 set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
 set(CMAKE_CONFIGURATION_TYPES Debug Release RelWithDebInfo CACHE STRING "" FORCE)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
For some reason, clang was not listening to the c++ standards set elsewhere. This was a quick and dirty way of telling clang to use c++17 for all of the projects.

I imagine we can find a cleaner and more proper way of doing this.

Reply:
This is now added by default, there is no reason for us to not have the entire project build as c++17